### PR TITLE
82 - Decoding of messageHeader and amqpValue

### PR DESCRIPTION
**Pull request by l4mby**
The feature is only for the decoding of these message parts in accordance with protocol amqp 1.0
The tests were done using a binary file containing the message to decode with messageHeader and amqpValue
Ref #82 #83

### CODE REVIEW

File: src/producer.ts [hunk 1]
```diff
@@ -94,6 +105,7 @@ export class Producer {
             message: {
               content: message,
               messageProperties: opts.messageProperties,
+              messageHeader: opts.messageHeader,
```
Comment (gpad) [inline src/producer.ts:108]: Could the header be sent? are you sure?

File: src/response_decoder.ts [hunk 2]
```diff
@@ -203,14 +206,49 @@ function decodeMessage(dataResponse: DataReader, offset: bigint): Message {
...
+        break
+      default:
+        throw new Error(`Not supported format code ${formatCode}`)
+    }
+  }
+
+  return { content, messageProperties, messageHeader, applicationProperties, amqpValue, offset }
+}
+
+export function decodeMessageTesting(dataResponse: DataReader, length: number): Message {
```
Comment (gpad) [inline src/response_decoder.ts:222]: What is this function? If it's used in test should be declared in test

File: src/response_decoder.ts [hunk 3]
```diff
@@ -183,12 +184,14 @@ function decodeDeliverResponse(dataResponse: DataReader, logger: Logger): Delive
 
 const EmptyBuffer = Buffer.from("")
 
-function decodeMessage(dataResponse: DataReader, offset: bigint): Message {
+export function decodeMessage(dataResponse: DataReader, offset: bigint): Message {
```
Comment (gpad) [inline src/response_decoder.ts:187]: why export?

File: src/response_decoder.ts [hunk 4]
```diff
@@ -238,6 +276,23 @@ function decodeMessageProperties(dataResponse: DataReader) {
   return Properties.parse(dataResponse, propertiesLength as number)
 }
 
+export function decodeMessageHeader(dataResponse: DataReader) {
```
Comment (gpad) [inline src/response_decoder.ts:279]: why export?

File: test/support/util.ts [hunk 5]
```diff
@@ -89,5 +89,13 @@ export async function createClassicConsumer(
 
   return { conn, ch }
 }
+
+export async function createClassicPublisher(): Promise<{ conn: ampq.Connection; ch: ampq.Channel }> {
+  const conn = await ampq.connect(`amqp://${username}:${password}@localhost`)
+  const ch = await conn.createChannel()
+
+  return { conn, ch }
+}
```
Comment (gpad) [inline test/support/util.ts:98]:
```suggestion
export async function createClassicPublisher(): Promise<{ conn: ampq.Connection; ch: ampq.Channel }> {
  const conn = await ampq.connect(`amqp://${username}:${password}@localhost`)
  const ch = await conn.createChannel()
  return { conn, ch }
}
```

File: src/producer.ts (hunk 1 above)
Comment (l4mby) [reply src/producer.ts:108]: Sorry my bad, I added this before I knew it shouldn't be sent. I will fix this

**Review by albertobarrila — APPROVED** [review]
all previous comments has been resolved so looks good to me

**Review by gpad — APPROVED** [review]

**Pull request merged**